### PR TITLE
[Backport 2.3-develop] CMS Block/Page admin improvements

### PR DESCRIPTION
**Pull request by adrian-martinez-interactiv4**
CMS Admin Improvements
Related with [PR#11836](https://github.com/magento/magento2/pull/11836)

### Description
This PR will:
- Use CMS Block repository to retrieve/save/delete data, instead of model load/save/delete
- Use CMS Page repository to retrieve/save/delete data, instead of model load/save/delete
- Remove deprecated calls to $messageManager
- Update CMS Page and Block unit tests, also covering throwing NoSuchEntityException and more cases than before

### Fixed Issues (if relevant)
None, code improvement

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by dmanners** [comment]
@adrian-martinez-interactiv4 can you work on the conflicting file and then I will start to process the 2.3 version of this PR first.

**Comment by dmanners** [comment]
@adrian-martinez-interactiv4 what do you think about adding a new service handler that the controllers will call? Currently the `MassDisable` and `MassEnable` controllers are being flagged by `Magento\Test\Php\LiveCodeTest::testCopyPaste` and since this PR is a refactoring PR maybe we take a bit of time here to extract everything we can.

What are your thoughts?

**Comment by adrian-martinez-interactiv4** [comment]
@dmanners my thought is that you are right, I'll give it a try

**Comment by adrian-martinez-interactiv4** [comment]
@dmanners I have finally unified functionality in the same controller, and updated tests. Please check https://github.com/magento/magento2/pull/11856/commits/d47fa4566e2b86b8db97dc8e02cbeed546a7d651

**Comment by dmanners** [comment]
Thanks for the update @adrian-martinez-interactiv4 I will take a look at this for you.

File: app/code/Magento/Cms/Controller/Adminhtml/Page/MassStatus.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     * Authorization level of a basic admin session
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Magento_Cms::save';
+
+    /**
+     * @var Filter
+     */
+    protected $filter;
```
Comment (dmanners) [inline app/code/Magento/Cms/Controller/Adminhtml/Page/MassStatus.php:31]: Since this is a new class we can update these from protected.

**Review by dmanners — CHANGES_REQUESTED** [review]

**Comment by dmanners** [comment]
@adrian-martinez-interactiv4 I am a little concerned with the removal of the controllers. Though I do prefer having 1 controller to deal with enable/disable similar to https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Catalog/Controller/Adminhtml/Product/MassStatus.php I wonder if we would need to consider a url rewrite or maybe even taking the logic out of the controller into a share class but keeping the two controllers. Happy to discuss further.

File: app/code/Magento/Cms/Controller/Adminhtml/Page/MassStatus.php (hunk 1 above)
Comment (adrian-martinez-interactiv4) [reply app/code/Magento/Cms/Controller/Adminhtml/Page/MassStatus.php:31]: Sure, but it only makes sense if we are going to maintain new controller class, let's discuss about it in the main thread

**Comment by dmanners** [comment]
Hi @adrian-martinez-interactiv4 sorry but I am going to close this PR. For me I feel this, though a good addition, is too many changes in one PR. I would love to see 3 different PRs.

1. To cover the repository usage instead of the model usage,
2. Remove the deprecated calls to `$messageManager`,
3. Update of the unit tests to cover more cases,

I would have sorted this out myself but unfortunately as it was all done in 1 commit I could not split out the changes while still giving you author credit.

If you feel like still make these changes please reach out and I would be happy to help.

Working with either smaller pull requests or multiple commits for different features/changes makes it a lot easier to review, test and approve.

Thank you for understanding.

**Comment by adrian-martinez-interactiv4** [comment]
@dmanners I understand there are many changes here, but maybe someone else could afford the responsibility to check proposed changes, so only as a possibility, couldn't be this PR reopened and reassigned to that someone else instead to throw all this work away?

**Comment by okorshenko** [comment]
Hi @adrian-martinez-interactiv4 
we will review this PR one more time

**Comment by dmanners** [comment]
@adrian-martinez-interactiv4 we have had a few PRs come in for smaller refactoring of this part of the admin section. With that in mind I will close this PR. If you feel that you would like to still provide some of this code, feel free to rebase it onto the latest 2.3-develop branch and reopen a pull request.

Thank you

**Pull request closed without merging**